addons/idporten.md

```diff
@@ -0,0 +1,339 @@
+using the [OAuth 2.0 Token Exchange].
+
+See the [TokenX Documentation] for NAV-specific usage.
+
+For more information about integration with provider see [ID-porten Integration guide].
+
+### Getting Started
+
+#### Minimal Example
+
+The very minimal example configuration required in [`nais.yaml`](../nais-application/reference.md#spec-idporten)
+to enable auto-provisioning of an ID-porten client for your application.
+
+```yaml
+apiVersion: "nais.io/v1alpha1"
+kind: "Application"
+metadata:
+  name: nais-testapp
+  namespace: aura
+  labels:
+    team: aura
+spec:
+  image: navikt/nais-testapp:66.0.0
+  idporten:
+    enabled: true
+  # required for on-premises only
+  webproxy: true
+  # required for GCP only
+  accessPolicy:
+    outbound:
+      external:
+        - host: oidc-ver2.difi.no
+```
+
+This will register an ID-porten client using the following naming scheme: 
+```
+<cluster>:<metadata.namespace>:<metadata.name>
+```
+
+For the example above, the result would be:
+```
+dev-gcp:aura:nais-testapp
+```
+
+#### Full Example
+
+```yaml
+apiVersion: "nais.io/v1alpha1"
+kind: "Application"
+metadata:
+  name: nais-testapp
+  namespace: aura
+  labels:
+    team: aura
+spec:
+  image: navikt/nais-testapp:66.0.0
+  idporten:
+    enabled: true
+    clientName: NAV - DEV
+    clientURI: "https://nav.no"
+    redirectURI: "https://my.application.dev.nais.io/callback"
+    frontchannelLogoutURI:
+      - "https://my.application.dev.nais.io/logout"
+    postLogoutRedirectURIs:
+      - "https://nav.no"
+    # refresh token last for 6 hours
+    refreshTokenLifetime: 21600
+  ingresses:
+    - "https://my.application"
+    - "https://my.application.dev.nais.io"
+  accessPolicy:
+    inbound:
+      rules:
+        - application: app-a
+          namespace: othernamespace
+        - application: app-b
+    # required for GCP only
+    outbound:
+      external:
+        - host: oidc-ver2.difi.no
+  # required for on-premises only
+  webproxy: true
+```
+
+### Redirect URI
+
+{% hint style="info" %}
+Note that `spec.idporten.redirectURI` can be omitted **if** a single `spec.ingresses` are specified.
+{% endhint %}
+
+The ingress specified will generate a redirect URL using the formula:
+
+```
+spec.ingresses + "/oauth2/callback"
+```
+
+In other words, this:
+
+```yaml
+spec:
+  idporten:
+    enabled: true
+  ingresses:
+    - "https://my.application.dev.nais.io"
+```
+
+is equivalent to this:
+
+```yaml
+spec:
+  idporten:
+    enabled: true
+    redirectURI: "https://my.application.dev.nais.io/oauth2/callback"
+  ingresses:
+    - "https://my.application.dev.nais.io"
+```
+
+You may override the redirect URL manually by specifying `spec.idporten.redirectURI`. 
+
+If you for any reason have several ingresses
+```yaml
+spec:
+  idporten:
+    enabled: true
+  ingresses:
+    - "https://my.application.dev.nais.io"
+    - "https://my.application/"
+```
+
+Specifying `spec.idporten.redirectURI` is **mandatory** in your nais.yml
```

Comment:
Er det kun lov med en redirect-url?

Reply:
Ja, her får jeg be @tommytroen eller @tronghn avare for meg 😁

Reply:
Går for kun én for nå, så kan vi tilpasse oss senere hvis det er behov for det.